src/aapns/api.py

```diff
@@ -75,33 +22,33 @@ class APNS:
         topic: Optional[str] = None,
         collapse_id: Optional[str] = None,
     ) -> str:
-        url, request_headers, request_body = encode_request(
-            token=token,
-            notification=notification,
-            apns_id=apns_id,
-            expiration=expiration,
-            priority=priority,
-            topic=topic,
-            collapse_id=collapse_id,
-            server=self.server,
-        )
 
-        response = await self.client.post(
-            url, data=request_body, headers=request_headers
+        r = Request.new(
+            path=f"/3/device/{token}",
+            header={
+                "apns-priority": str(priority.value),
+                "apns-push-type": notification.push_type.value,
+                **({"apns-id": apns_id} if apns_id else {}),
+                **({"apns-expiration": str(expiration)} if expiration else {}),
+                **({"apns-topic": topic} if topic else {}),
+                **({"apns-collapse-id": collapse_id} if collapse_id else {}),
+            },
+            data=notification.get_dict(),
+            timeout=10,
         )
-
-        return await handle_response(response)
+        return (await self.pool.post(r)).apns_id
 
     async def close(self):
-        await self.client.aclose()
+        await self.pool.close()
```

Comment:
if the pool has to be passed in created, I think this method should be removed and the caller has to close the pool. otherwise creation/teardown is asymetric.